x-pack/plugins/secops/public/components/formatted_date/index.test.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mount } from 'enzyme';
+import moment from 'moment-timezone';
+import * as React from 'react';
+
+import { KibanaConfigContext, PreferenceFormattedDate } from '.';
+import { AppFrameworkAdapter } from '../../lib/lib';
+import { mockFrameworks } from '../../mock';
+
+describe('PreferenceFormattedDate', () => {
+  describe('rendering', () => {
+    const isoDateString = '2019-02-25T22:27:05.000Z';
+    const configFormattedDateString = (
+      dateString: string,
+      config: Partial<AppFrameworkAdapter>
+    ): string => moment.tz(dateString, config.dateFormatTz!).format(config.dateFormat);
```

Comment:
Is that right? You made it partial and then used a `!` right afterwards for it? :-) It's tests so I expect to see some interesting things with TypeScript.

Reply:
So the test data is a Partial of the `AppFrameworkAdapter` as it's not necessary to mock out the entire object. Should I create a sub-type for testing that doesn't have `config.dateFormaTz` as an optional, or is the `!` okay in this instance? What's best practice here?

Reply:
As an aside, it looks like there exists an `AppTestingFrameworkAdapter` class that I can reference instead, but it still has the same typing for those fields.